GitHubTeams.ps1

```diff
@@ -276,6 +314,407 @@ filter Get-GitHubTeamMember
+        Justification = 'Methods called within here make use of PSShouldProcess, and the switch
+        is passed on to them inherently.')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', '',
+        Justification = 'One or more parameters (like NoStatus) are only referenced by helper
+        methods which get access to it from the stack via Get-Variable -Scope 1.')]
+    [CmdletBinding(
+        SupportsShouldProcess,
+        PositionalBinding = $false
+    )]
+    param
+    (
+        [Parameter(Mandatory, Position = 1)]
+        [ValidateNotNullOrEmpty()]
+        [string] $OrganizationName,
+
+        [Parameter(Mandatory, Position = 2)]
+        [ValidateNotNullOrEmpty()]
+        [string] $TeamName,
+
+        [string] $Description,
+
+        [string[]] $Maintainers,
+
+        [string[]] $RepositoryName,
+
+        [ValidateSet('Secret','Closed')]
+        [string] $Privacy,
+
+        [string] $ParentTeamName,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $telemetryProperties = @{
+        OrganizationName = (Get-PiiSafeString -PlainText $OrganizationName)
+        TeamName = (Get-PiiSafeString -PlainText $TeamName)
+    }
+
+    $uriFragment = "/orgs/$OrganizationName/teams"
+
+    $hashBody = @{
+        name = $TeamName
+    }
+
+    if ($PSBoundParameters.ContainsKey('Description')) { $hashBody['description'] = $Description }
+    if ($PSBoundParameters.ContainsKey('Maintainers')) { $hashBody['maintainers'] = $Maintainers }
+    if ($PSBoundParameters.ContainsKey('RepositoryName')) { $hashBody['repo_names'] = $RepositoryName }
+    if ($PSBoundParameters.ContainsKey('Privacy')) { $hashBody['privacy'] = $Privacy.ToLower() }
+    if ($PSBoundParameters.ContainsKey('ParentTeamName'))
+    {
+        $getGitHubTeamParms = @{
+            OrganizationName = $OrganizationName
+            TeamName = $ParentTeamName
+            Whatif = $false
+            Confirm = $false
+        }
+        if ($PSBoundParameters.ContainsKey('AccessToken'))
+        {
+            $getGitHubTeamParms['AccessToken'] = $AccessToken
+        }
+        if ($PSBoundParameters.ContainsKey('NoStatus'))
+        {
+            $getGitHubTeamParms['NoStatus'] = $NoStatus
+        }
+        $team = Get-GitHubTeam @getGitHubTeamParms
+
+        $hashBody['parent_team_id'] = $team.id
+    }
+
+    $params = @{
+        UriFragment = $uriFragment
+        Body = (ConvertTo-Json -InputObject $hashBody)
+        Method = 'Post'
+        Description =  "Creating $TeamName"
+        AccessToken = $AccessToken
+        TelemetryEventName = $MyInvocation.MyCommand.Name
+        TelemetryProperties = $telemetryProperties
+        NoStatus = (Resolve-ParameterWithDefaultConfigurationValue `
+            -Name NoStatus -ConfigValueName DefaultNoStatus)
+    }
+
+    return Invoke-GHRestMethod @params
+}
+
+function Update-GitHubTeam
```

Comment:
please update to be a `filter` and all other relevant pipeline-supporting changes.  This would definitely benefit from filter input, being able to pass-in a Team object...which means that the slug needs to be an optional parameter set.